### DISCUSSION TITLE
Code highlighting

### DISCUSSION THREAD

**Discussion by ddkwork** [cogentcore/core#880] [Q&A] [answered]
Code highlighting: Hello, I looked at the demo of the text editor and part of the code app, and found that the lang field of the BUF cannot be syntax highlighted, can you write a shortened example? If there's a way to customize the keyword highlight color, I'd like to highlight this debugger script https://github.com/HyperDbg/scripts/blob/master/windows%2FNetwork%2Fprocess-specific-connections-ip-port.ds

**Comment by kkoreilly** [comment]
Have you tried setting the `texteditor.Buf.Hi.Lang`?

**Comment by ddkwork** [comment]
Tried it



---Original---
From: ***@***.***&gt;
Date: Thu, Feb 15, 2024 23:43 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [cogentcore/core] Code highlighting (Discussion #880)




 
Have you tried setting the texteditor.Buf.Hi.Lang?

**Reply by kkoreilly** [reply]
What did you set it to? We probably don't have support for this specific debugger script language, so can you try picking a language that it resembles, like C?

**Reply by ddkwork** [reply]
> What did you set it to? We probably don't have support for this specific debugger script language, so can you try picking a language that it resembles, like C?

```go
fn.Buf.Hi.Style = "emacs"
fn.Buf.Hi.Lang = "Go"
```

**Reply by kkoreilly** [reply]
As I said, try setting it to a language more like the debugger script language, like C.

**Reply by ddkwork** [reply]
> As I said, try setting it to a language more like the debugger script language, like C.

The test go code is also not highlighted

**Reply by kkoreilly** [reply]
Are you using the latest version of core? I fixed a highlighting bug around a week ago.

**Reply by ddkwork** [reply]
I'll try again

**Reply by ddkwork** [reply]
![screenshots](https://github.com/cogentcore/core/assets/19886504/c91b69c9-99a5-4eb6-82a6-5754891274ad)

**Reply by kkoreilly** [reply, marked as answer]
Apologies for the issue; there was a problem with the file opening logic in the text editor. It should be fixed now if you pull to the latest version; it should automatically highlight both the Go code and the debug script.

**Comment by ddkwork** [comment]
Hello, it might be too cumbersome to make a disassembly display widget, is it feasible to add keyword highlighting to each cell of the table? Can you write a demo, thanks
![image](https://github.com/cogentcore/core/assets/19886504/f87a21b2-2159-4fcc-a995-1f4ed3289736)
https://ollydbg.de/

**Reply by kkoreilly** [reply]
I will look into this.

**Reply by kkoreilly** [reply]
I would recommend using a text editor for this, and you can set the language to anything supported by https://github.com/alecthomas/chroma, or you could write your own lexer if none of those work. I can help you if you need.

**Reply by ddkwork** [reply]
> I would recommend using a text editor for this, and you can set the language to anything supported by https://github.com/alecthomas/chroma, or you could write your own lexer if none of those work. I can help you if you need.

Your idea is very creative, but how should the asm assembly instruction highlight it, in addition, each line of assembly code needs to be line selected, jump instruction folding, column-splitting lines similar to tables, etc., for me tree tables, disassembly display, memory editor, these are too difficult to do

**Reply by kkoreilly** [reply]
We will implement tree tables in #675, so you should be able to implement this after we do that. I am planning to work more on that after I finish some other things that affect all apps.

**Reply by ddkwork** [reply]
> We will implement tree tables in #675, so you should be able to implement this after we do that. I am planning to work more on that after I finish some other things that affect all apps.

thanks

**Comment by ddkwork** [comment]
Thank you for your pointers, I'll take the time to try this idea



---Original---
From: ***@***.***&gt;
Date: Sun, Feb 18, 2024 09:24 AM
To: ***@***.***&gt;;
Cc: ***@***.***&gt;;"State ***@***.***&gt;;
Subject: Re: [cogentcore/core] Code highlighting (Discussion #880)




 
I would recommend using a text editor for this, and you can set the language to anything supported by https://github.com/alecthomas/chroma, or you could write your own lexer if none of those work. I can help you if you need.

**Comment by ddkwork** [comment]
![screenshots](https://github.com/cogentcore/core/assets/19886504/4a89ff01-e215-40f4-889f-5bd5db76ad70)
